core/src/autogluon/core/searcher/bayesopt/models/gpmodel_transformers.py

```diff
@@ -200,24 +216,58 @@ def _compute_model(self, skip_optimization=None):
                 logger.info(
                     "Skipping the refitting of GP hyperparameters, since the "
                     "labeled data did not change since the last recent fit")
-        self._model = GaussProcSurrogateModel(
-            state=self._state,
-            active_metric=args.active_metric,
-            random_seed=args.random_seed,
-            gpmodel=self._gpmodel,
-            fit_parameters=fit_parameters,
-            num_fantasy_samples=args.num_fantasy_samples,
-            normalize_targets=args.normalize_targets,
-            profiler=self._profiler,
-            debug_log=self._debug_log,
-            debug_fantasy_values=self._debug_fantasy_values)
-        # DEBUG: Supplied values are used only once
-        self._debug_fantasy_values = None
+
+        if self._use_single_model:
```

Comment:
nitpick:
This could be done without copy&paste, something like this:

if self._use_single_model:
   gpmodels = dictionarize_objective(self._gpmodel)
else:
   gpmodels = self._gpmodel
output_models = {
   GaussProcSurrogateModel(...)
   for metric_name, gpmodel in gpmodels
}
if self._use_single_model:
   self._model = output_models[self.active_metric]
else:
   self._model = output_models

You are using dictionarize_objective elsewhere for this. That's elegant, but maybe the name is misleading then.

Reply:
Actually, please do this. We don't want c&p.